packages/react-components/src/components/AppFrame/components/NavigationTopBar/types.ts

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+
+import { ComponentCoreProps } from 'utils/types';
+
+export type NavigationTopBarKind = 'info' | 'success' | 'warning' | 'error';
+
+export interface INavigationTopBarProps extends ComponentCoreProps {
+  /**
+   * Contents of the top bar. You can use the `NavigationTopBar.Alert` component to display alerts.
+   */
+  children: React.ReactNode;
+
+  /**
+   * Title of the top bar. This is optional, and the title will be rendered below the children if defined.
+   */
+  title?: React.ReactNode;
+}
+
+export interface ITopBarAlertProps extends ComponentCoreProps {
+  /**
+   * Visual style of the alert. You can also use `className` to style the alert if you need to set a custom background color.
+   * Defaults to `info`.
+   * */
+  kind?: NavigationTopBarKind;
+  /**
+   * Properties of the close button. If defined, the alert will be closable and the close button will be rendered.
+   * aria-label is highly recommended for accessibility.
+   * */
+  closeButton?: {
+    onClick: () => void;
+    'data-testid'?: string;
+    'aria-label'?: string;
+  };
+  /**
+   * Content of the alert.
+   */
+  children: React.ReactNode;
+
+  /**
+   * Primary CTA button. The button will be rendered if defined.
+   * Simple on purpose - if you need more complex buttons, you can use the `children` prop.
+   * */
+  primaryCta?: {
+    label: string;
+    onClick: () => void;
+  };
+
+  /**
+   * Secondary CTA button. The button will be rendered if defined.
+   * Simple on purpose - if you need more complex buttons, you can use the `children` prop.
+   * */
+  secondaryCta?: {
+    label: string;
+    onClick: () => void;
+  };
+
+  /**
+   * Show or hide the alert, defaults to `true`. Changes to this prop are animated - the alert will enter and leave smoothly.
+   * */
+  show?: boolean;
```

Comment:
Historically, our codebase commonly uses `isVisible` naming. Therefore, I suggest using similar naming to maintain a consistent pattern.
P.S.: Consistent naming patterns are something I highly desire, but for various reasons, we don't have them yet. 😄

Reply:
> but for various reasons, we don't have them yet. 😄

sorry to have been one of these reasons! it's `isVisible` now

Reply:
@ashbork Haha, no, I meant legacy code from the previous DS version, since part of the new API is a clone of the old one. 😄 
You and the other contributors are actually the reason why the codebase keeps improving, so great job and thanks for your effort! 🏆